mantis-runtime-loader/src/main/java/io/mantisrx/runtime/loader/SubscriptionStateHandlerImpl.java

```diff
@@ -53,7 +53,9 @@ class SubscriptionStateHandlerImpl extends AbstractScheduledService implements S
 
     @Override
     public void startUp() {
-        currentState.set(SubscriptionState.of(clock));
+        if (currentState.get() == null) {
```

Comment:
seems like you are expecting scenarios where startUp can be called multiple times?

Reply:
The currentState can be set at startUp or the first update from subscribe event (currently there is no set at subscribe so if subscribe happens before startup it hits NPE).

Reply:
We need to ensure that none of the members of the class are called before startUp, as this would violate the semantics of a Service class. To prevent this, I suggest setting up preconditions in the listener methods rather than working around the issue.

Reply:
I think it might be a better idea to start the subscription handler in
```
private void setupSubscriptionStateHandler(ExecuteStageRequest executeStageRequest) {
```